src/runtime/cpu/loss/loss.cpp

```diff
@@ -0,0 +1,50 @@
+#include <vector>
+
+#include "loss.hpp"
+
+namespace deepworks {
+
+void deepworks::CrossEntropyLoss::CPUCrossEntropyLossForward(const ConstMatrix &X,
+                                                             const ConstVector &target,
+                                                             Matrix &loss) {
+    int rows = X.rows();
+    int cols = X.cols();
+
+    Eigen::MatrixXf pr(rows, cols);
+    Matrix P(pr.data(), rows, cols);
+
+    CPUSoftmaxForward(X, P);
```

Comment:
@l-bat Should it be there like `pytorch` or we are going to do it outside the loss ?

Reply:
I believe it should be outside the class, because we have `Softmax` as a layer in API

Reply:
Possible it should be names `NLLLoss` and used after `dw::Softmax` to make this logic clearer

Reply:
For this implementation  of `CPUCrossEntropyLossForward` we should create a model without `dw::Softmax` layer. It may cause a problem on evaluation, because users should compute  `Softmax` manually. 
My suggestion is computing log + NLLLoss in separate functions:
```cpp
float CrossEntropyLoss::CPUCrossEntropyLossForward(const Tensor& X, const Tensor& target) {
    // map Tensor to ConstMatrix XMat
   CPULog(XMat, XMat);
   CPUNLLLoss(XMat, targetVector);
}
```